samples/identity/credentials/src/default-azure-credential-debug-logger.ts

```diff
@@ -0,0 +1,62 @@
+// <DEFAULT_AZURE_CREDENTIAL_DEBUG>
+import { setLogLevel, AzureLogger } from "@azure/logger";
+import { BlobServiceClient } from "@azure/storage-blob";
+import { 
+    DefaultAzureCredential 
+} from "@azure/identity";
+
+// Constant for the Azure identity log prefix
+const AZURE_IDENTITY_LOG_PREFIX = "azure:identity";
+
+// override logging to output to console.log (default location is stderr)
+// only log messages that start with the Azure identity log prefix
+setLogLevel("verbose");
+AzureLogger.log = (...args) => {
+  const message = args[0];
+  if (typeof message === 'string' && message.startsWith(AZURE_IDENTITY_LOG_PREFIX)) {
+    console.log(...args);
+  }
+};
+
+// Get storage account name from environment variable
+const storageAccountName = process.env.AZURE_STORAGE_ACCOUNT_NAME;
+
+if (!storageAccountName) {
+    throw new Error("AZURE_STORAGE_ACCOUNT_NAME environment variable is required");
+}
+
+const credential = new DefaultAzureCredential();

```

Comment:
DAC should enforce that the appropriate env var is actually defined. You can do that as follows:

```suggestion
const credential = new DefaultAzureCredential({ 
    requiredEnvVars: [ "AZURE_TOKEN_CREDENTIALS" ]
});

```